tests/beaker_tests/cisco_itd_device_group/test_itd_device_group.rb

```diff
@@ -0,0 +1,161 @@
+###############################################################################
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+#
+# See README-develop-beaker-scripts.md (Section: Test Script Variable Reference)
+# for information regarding:
+#  - test script general prequisites
+#  - command return codes
+#  - A description of the 'tests' hash and its usage
+#
+###############################################################################
+
+require File.expand_path('../../lib/utilitylib.rb', __FILE__)
+
+# Test hash top-level keys
+tests = {
+  master:           master,
+  agent:            agent,
+  operating_system: 'nexus',
+  resource_name:    'cisco_itd_device_group',
+}
+
+# Test hash test cases
+tests[:default] = {
+  desc:           '1.1 Defaults',
+  title_pattern:  'icmpGroup',
+  preclean:       'cisco_itd_device_group',
+  manifest_props: {
+    probe_type: 'default'
+  },
+  code:           [0, 2],
+  resource:       {
+    'probe_type' => 'false'
+  },
+}
+
+tests[:default_icmp] = {
```

Comment:
Don't remove the `platform:` key.  The presence of this key restricts tests to platforms in the list so instead I would just replace `operating_system: 'nexus'` since this is only supported on `n7k` and `n9k` and replace it with `platform: 'n(7|9)k'`.